[V2] az-log panics when getting the controller logs

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->


**What happened**:

`az-log` panics when getting the controller logs:

```log
panic: leases.coordination.k8s.io "default" not found
goroutine 1 [running]:
sigs.k8s.io/azuredisk-csi-driver/cmd/az-log/cmd.GetLeaderControllerPod({0x1a61840, 0xc000312000})
        /home/jlong/azuredisk-csi-driver/cmd/az-log/cmd/controller.go:69 +0x385
sigs.k8s.io/azuredisk-csi-driver/cmd/az-log/cmd.glob..func1(0x25a2c80?, {0x1805b2c?, 0x2?, 0x2?})
        /home/jlong/azuredisk-csi-driver/cmd/az-log/cmd/controller.go:43 +0xe5
github.com/spf13/cobra.(*Command).execute(0x25a2c80, {0xc000357b40, 0x2, 0x2})
        /home/jlong/azuredisk-csi-driver/vendor/github.com/spf13/cobra/command.go:876 +0x67b
github.com/spf13/cobra.(*Command).ExecuteC(0x25a3900)
        /home/jlong/azuredisk-csi-driver/vendor/github.com/spf13/cobra/command.go:990 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/jlong/azuredisk-csi-driver/vendor/github.com/spf13/cobra/command.go:918
sigs.k8s.io/azuredisk-csi-driver/cmd/az-log/cmd.Execute()
        /home/jlong/azuredisk-csi-driver/cmd/az-log/cmd/root.go:43 +0x25
main.main()
        /home/jlong/azuredisk-csi-driver/cmd/az-log/main.go:22 +0x17


goroutine 41 [IO wait]:
internal/poll.runtime_pollWait(0x7fe060619678, 0x72)
        /usr/local/go/src/runtime/netpoll.go:305 +0x89
internal/poll.(*pollDesc).wait(0xc00037e880?, 0xc000594a80?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x32
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc00037e880, {0xc000594a80, 0xa80, 0xa80})
        /usr/local/go/src/internal/poll/fd_unix.go:167 +0x25a
net.(*netFD).Read(0xc00037e880, {0xc000594a80?, 0xc00013ba60?, 0xc000594b67?})
        /usr/local/go/src/net/fd_posix.go:55 +0x29
net.(*conn).Read(0xc000014008, {0xc000594a80?, 0x50000c0004b17e0?, 0x40f4e7?})
        /usr/local/go/src/net/net.go:183 +0x45
crypto/tls.(*atLeastReader).Read(0xc00012e258, {0xc000594a80?, 0x0?, 0x6fa1f9?})
        /usr/local/go/src/crypto/tls/conn.go:787 +0x3d
bytes.(*Buffer).ReadFrom(0xc0003c85f8, {0x1a38360, 0xc00012e258})
        /usr/local/go/src/bytes/buffer.go:202 +0x98
crypto/tls.(*Conn).readFromUntil(0xc0003c8380, {0x1a3b9e0?, 0xc000014008}, 0x99e?)
        /usr/local/go/src/crypto/tls/conn.go:809 +0xe5
crypto/tls.(*Conn).readRecordOrCCS(0xc0003c8380, 0x0)
        /usr/local/go/src/crypto/tls/conn.go:616 +0x116
crypto/tls.(*Conn).readRecord(...)
        /usr/local/go/src/crypto/tls/conn.go:582
crypto/tls.(*Conn).Read(0xc0003c8380, {0xc00053c000, 0x1000, 0xa8e2a0?})
        /usr/local/go/src/crypto/tls/conn.go:1287 +0x16f
bufio.(*Reader).Read(0xc0002da060, {0xc00015c2e0, 0x9, 0xa9c5c5?})
        /usr/local/go/src/bufio/bufio.go:237 +0x1bb
io.ReadAtLeast({0x1a38200, 0xc0002da060}, {0xc00015c2e0, 0x9, 0x9}, 0x9)
        /usr/local/go/src/io/io.go:332 +0x9a
io.ReadFull(...)
        /usr/local/go/src/io/io.go:351
golang.org/x/net/http2.readFrameHeader({0xc00015c2e0?, 0x9?, 0xc001590330?}, {0x1a38200?, 0xc0002da060?})
        /home/jlong/azuredisk-csi-driver/vendor/golang.org/x/net/http2/frame.go:237 +0x6e
golang.org/x/net/http2.(*Framer).ReadFrame(0xc00015c2a0)
        /home/jlong/azuredisk-csi-driver/vendor/golang.org/x/net/http2/frame.go:498 +0x95
golang.org/x/net/http2.(*clientConnReadLoop).run(0xc0004b1f98)
        /home/jlong/azuredisk-csi-driver/vendor/golang.org/x/net/http2/transport.go:2118 +0x130
golang.org/x/net/http2.(*ClientConn).readLoop(0xc000140480)
        /home/jlong/azuredisk-csi-driver/vendor/golang.org/x/net/http2/transport.go:2014 +0x6f
created by golang.org/x/net/http2.(*Transport).newClientConn
        /home/jlong/azuredisk-csi-driver/vendor/golang.org/x/net/http2/transport.go:725 +0xaaa
```

**What you expected to happen**:

It should return the log for the current leader.

**How to reproduce it**:

**Anything else we need to know?**:

`az-log` assumes the lease name for the CSI controller is "default" which is no longer true. The current default value is `csi-azuredisk-controller` but can be overridden through the `--controller-partition` command-line flag or the `AzDriverConfiguration.ControllerConfig.PartitionName` configuration value. The lease is in the namespace defined by the `--leader-election-namespace` command-line flag or the `AzDriverConfiguration.ControllerConfig.LeaderElectionNamespace` configuration value.

The change was made by PR #1363.

**Environment**:
<!-- 
*Run following command to get CSI driver version:
kubectl get po -n kube-system -o yaml | grep mcr | grep azuredisk
-->
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/cmd/az-log/README.md
+++ b/cmd/az-log/README.md
@@ -3,12 +3,6 @@
 ## About
 az-log is a command line tool to fetch and parse appropriate logs from the driver plugins to track the operation workflow end to end and bring better insight of driver behaviors.
 
-## Installation
-```console
-$ cd $GOPATH/src/sigs.k8s.io/azuredisk-csi-driver
-$ make install-az-log
-```
-
 ## Configuration
 Create an az-log config file in one of the three paths below:
 ```console
@@ -28,7 +22,16 @@ kubeconfig: "" # default is "$HOME/.kube/config"
 driverNamespace: "" # default is "azure-disk-csi"
 releaseNamesapce: "" # default is "kube-system"
 v2InstalledSideBySide: "" # default is "false"
+leaderElectionNamespace: "" # default is "kube-system"
+leaseName: "" # default is "csi-azuredisk-controller"
 ```
+
+## Installation
+```console
+$ cd $GOPATH/src/sigs.k8s.io/azuredisk-csi-driver
+$ make install-az-log
+```
+
 ## Features
 
 ### Source Options

--- a/cmd/az-log/cmd/common.go
+++ b/cmd/az-log/cmd/common.go
@@ -172,3 +172,17 @@ func GetDriverInstallationType() string {
 	}
 	return "csi-azuredisk2"
 }
+
+func GetLeaderElectionNamespaceAndLeaseName() (string, string) {
+	leaderElectionNamespace := viper.GetString("leaderElectionNamespace")
+	leaseName := viper.GetString("leaseName")
+
+	if leaderElectionNamespace == "" {
+		leaderElectionNamespace = consts.ReleaseNamespace
+	}
+
+	if leaseName == "" {
+		leaseName = consts.DefaultControllerPartitionName
+	}
+	return leaderElectionNamespace, leaseName
+}

--- a/cmd/az-log/cmd/controller.go
+++ b/cmd/az-log/cmd/controller.go
@@ -26,8 +26,9 @@ import (
 	"github.com/spf13/cobra"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/client-go/kubernetes"
-	consts "sigs.k8s.io/azuredisk-csi-driver/pkg/azureconstants"
 )
 
 // controllerCmd represents the controller command
@@ -48,6 +49,7 @@ var controllerCmd = &cobra.Command{
 			if !isFollow {
 				break
 			} else {
+				sinceTime = time.Now().Format(time.RFC3339)
 				time.Sleep(20 * time.Second)
 				pod = GetLeaderControllerPod(clientsetK8s)
 				if pod.Name == currPodName {
@@ -63,26 +65,31 @@ func init() {
 }
 
 func GetLeaderControllerPod(clientsetK8s kubernetes.Interface) *v1.Pod {
-	//Get node that leader controller pod is running in
-	lease, err := clientsetK8s.CoordinationV1().Leases(consts.DefaultAzureDiskCrdNamespace).Get(context.Background(), "default", metav1.GetOptions{})
+	// get LeaderElectionNamespace and PartitionName
+	leaderElectionNamespace, leaseName := GetLeaderElectionNamespaceAndLeaseName()
+
+	//get node that leader controller pod is running on
+	lease, err := clientsetK8s.CoordinationV1().Leases(leaderElectionNamespace).Get(context.Background(), leaseName, metav1.GetOptions{})
 	if err != nil {
 		panic(err.Error())
 	}
 	holder := *lease.Spec.HolderIdentity
 	node := strings.Split(holder, "_")[0]
 
-	// Get pod name of the leader controller
+	// get pod name of the leader controller
+	fieldSelector := fields.SelectorFromSet(map[string]string{"spec.nodeName": node})
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"app": GetDriverInstallationType() + "-controller"}}
 	pods, err := clientsetK8s.CoreV1().Pods(GetReleaseNamespace()).List(context.Background(), metav1.ListOptions{
-		FieldSelector: "spec.nodeName=" + node,
-		LabelSelector: "app=" + GetDriverInstallationType() + "-controller",
+		FieldSelector: fieldSelector.String(),
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
 	})
 	if err != nil {
 		panic(err.Error())
 	}
+
 	if len(pods.Items) != 1 {
 		fmt.Println("zero or more than one controller plugins were found")
 		os.Exit(0)
 	}
-
 	return &pods.Items[0]
 }

